python/lsst/daf/butler/direct_butler.py

```diff
@@ -1875,6 +1876,124 @@ def doImport(importStream: TextIO | ResourceHandleProtocol) -> None:
         else:
             doImport(filename)  # type: ignore
 
+    def transfer_dimension_records_from(
+        self, source_butler: LimitedButler | Butler, source_refs: Iterable[DatasetRef]
+    ) -> None:
+        # Allowed dimensions in the target butler.
+        elements = frozenset(
+            element for element in self.dimensions.elements if element.hasTable() and element.viewOf is None
+        )
+
+        data_ids = {ref.dataId for ref in source_refs}
+
+        dimension_records = self._extract_all_dimension_records_from_data_ids(
+            source_butler, data_ids, elements
+        )
+
+        for element, r in dimension_records.items():
```

Comment:
Does the order of element matter here and how would I work it out? Or am I supposed to open a transaction here?

Reply:
Ah, good catch; I think you want to iterate in `DimensionUniverse.sorted` order, which is also the same as `DimensionGroup` iteration order.